### PR TITLE
CAMEL-6707: add async for servlet

### PR DESCRIPTION
**Pull request by arnaud-deprez**
Add async option in servlet.

# Remarks:
I have some issues to test servlet with arquillian & jetty 9. Apparently it doesn't take into account the option async-supported in the web.xml (see test). But it works well with tomcat and the jetty component (because here we register the servlet ourself correctly).

### CODE REVIEW

File: components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java [hunk 1]
```diff
@@ -53,10 +59,61 @@
     public void init(ServletConfig config) throws ServletException {
         super.init(config);
         this.servletName = config.getServletName();
+        this.async = Optional.ofNullable(config.getInitParameter(ASYNC_PARAM))
```
Comment (davsclaus) [inline components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java]: Please avoid/abuse using optional for a simple check like this.

File: components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java [hunk 2]
```diff
@@ -16,35 +16,41 @@
...
-
 import org.apache.camel.Exchange;
 import org.apache.camel.ExchangePattern;
+import org.apache.camel.RuntimeCamelException;
 import org.apache.camel.impl.DefaultExchange;
+import org.apache.camel.util.ObjectHelper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.servlet.AsyncContext;
```
Comment (davsclaus) [inline components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java]: do not reorder the imports. Run mvn -Psourcecheck to validate the code

Comment (arnaud-deprez) [inline components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java]: Sorry I didn't pay attention, Intellij did it in my back, I need to update my IDEA config. I'll fix that and use the sourcecheck profile

File: components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java (hunk 1 above)
Comment (arnaud-deprez) [inline components/camel-http-common/src/main/java/org/apache/camel/http/common/CamelServlet.java]: ok

**Comment by arnaud-deprez** [comment]
Ok sourcecheck passed and I removed the Optional usages.
Rgds

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by arnaud-deprez** [comment]
Sure :)

**Pull request closed without merging**